File: core/core-api/src/main/java/com/coffee/gu/api/controller/v1/response/menu/MenuResponse.java
package com.coffee.gu.api.controller.v1.response.menu;

import com.coffee.gu.menu.Menu;

import java.math.BigDecimal;
import java.util.List;

public record MenuResponse(
    Long id,
    String name,
    BigDecimal salesPrice
) {
    public static List<MenuResponse> from(List<Menu> menus) {
        return menus.stream()
                .map(menu -> new MenuResponse(menu.getId(), menu.getName(), menu.getSalesPrice()))
                .toList();
    }
}

File: core/core-api/src/main/kotlin/com/coffee/gu/api/controller/v1/response/menu/MenuResponse.kt
package com.coffee.gu.api.controller.v1.response.menu

import com.coffee.gu.menu.Menu
import java.math.BigDecimal

class MenuResponse(
    val id: Long,
    val name: String,
    val salesPrice: BigDecimal,
) {
    companion object {
        fun from(menus: List<Menu>): List<MenuResponse> {
            return menus.map { menu ->
                MenuResponse(
                    id = menu.id,
                    name = menu.name,
                    salesPrice = menu.salesPrice
                )
            }
        }
    }
}
